internal/jobs/tiktok.go

```diff
@@ -260,6 +301,76 @@ func (ttt *TikTokTranscriber) ExecuteJob(j types.Job) (types.JobResult, error) {
 	return types.JobResult{Data: jsonData}, nil
 }
 
+// executeSearchByQuery runs the epctex/tiktok-search-scraper actor and returns results
+func (ttt *TikTokTranscriber) executeSearchByQuery(j types.Job, a *teeargs.TikTokSearchByQueryArguments) (types.JobResult, error) {
+	if ttt.configuration.ApifyApiKey == "" {
+		ttt.stats.Add(j.WorkerID, stats.TikTokErrors, 1)
+		return types.JobResult{Error: "Apify API key not configured for searchbyquery"}, fmt.Errorf("missing Apify API key")
+	}
+
+	c, err := tiktokapify.NewTikTokApifyClient(ttt.configuration.ApifyApiKey)
+	if err != nil {
+		ttt.stats.Add(j.WorkerID, stats.TikTokErrors, 1)
+		return types.JobResult{Error: "Failed to create Apify client"}, fmt.Errorf("apify client: %w", err)
+	}
+
+	limit := a.MaxItems
+	if limit <= 0 {
+		limit = 20
+	}
+
+	items, next, err := c.SearchByQuery(*a, client.EmptyCursor, limit)
+	if err != nil {
+		ttt.stats.Add(j.WorkerID, stats.TikTokErrors, 1)
```

Comment:
I just realized that I forgot to add stats to Reddit!!!

Reply:
haha and we'll soon forget we need to score the work on the validator side!  this is why I think we should have a "checklist" for new data sources - touches lots of parts of the stack

Reply:
I have the beginnings of one here: https://github.com/masa-finance/tee-indexer#adding-a-new-jobtype-andor-source, we should probably add to it. I think that this is the proper place since a) it's our private repo, and b) new datasources start in the tee-stack. I wouldn't put it in the tee-worker repo since it's public.